setup.py

```diff
@@ -10,46 +10,89 @@
         sys.exit(1)
 
-    cwd = os.path.abspath(os.path.dirname(__file__))
+    # NumPy include dir
+    numpy_include_dir = [ numpy.get_include() ]
 
-    # install required numpy
-    p = subprocess.call([sys.executable, "-m", 'pip', 'install', f'numpy{numpy_requires}'],cwd=cwd)
-    if p != 0:
-        # Could be due to a too old pip version and build isolation, check that
-        try:
-            # Note, pip may not be installed or not have been used
-            import pip
-            if LooseVersion(pip.__version__) < LooseVersion('18.0.0'):
-                raise RuntimeError("Installing requirements failed. Possibly due "
-                                   "to `pip` being too old, found version {}, "
-                                   "needed is >= 18.0.0.".format(pip.__version__))
-            else:
-                raise RuntimeError("Installing requirements failed!")
-        except ImportError:
-            raise RuntimeError("Installing requirement failed! `pip` has to be installed!")
-
-    # include dir
-    from numpy import get_include as numpy_get_include
-    numpy_include_dir = [numpy_get_include()]
-    
     # CFLAG
     # I intend to use -Ofast, however if gcc version < 4.6, this option is unavailable so...
-    extra_c_args = ["-w","-O3","-ffast-math","-g0"] # for C, -Ofast implies -O3 and -ffast-math
-    extra_c_args_for_fermi = extra_c_args + ["-std=gnu99","-DUSE_SIMDE", "-DSIMDE_ENABLE_NATIVE_ALIASES"]
-    
+    # should I care about old gcc compiler?...
+    extra_c_args = ["-w","-Ofast", "-g0"] # for C, -Ofast implies -O3 and -ffast-math
+
+    # CFLAG for fermi-lite related codes
+    clang = False
+    icc = False
+    new_gcc = False
+    try:
+        if os.environ['CC'] == "clang":
+            clang = True
+    except KeyError:
+        pass
+
+    if not clang:
+        try:
+            gcc_version_check = subprocess.check_output( ["gcc", "--version"], universal_newlines=True)
+            if gcc_version_check.find("clang") != -1:
+                clang = True
+            else:
+                gcc_version_check = gcc_version_check.split('\n')[0] # get the first line
+                m = re.search( "\s+(\d+\.\d+)\.\d+", gcc_version_check )
+                if m:
+                    gcc_version = float( m[1] )
+                    if gcc_version > 4.8:
+                        new_gcc = True
+        except subprocess.CalledProcessError:
+            pass
+
+    try:
+        if os.environ['CC'] == "icc":
+            icc = True
+    except KeyError:
+        pass
+
+    extra_c_args_for_fermi = ["-std=gnu99","-DUSE_SIMDE", "-DSIMDE_ENABLE_NATIVE_ALIASES"]
+    if icc or sysconfig.get_config_vars()['CC'] == 'icc':
+        extra_c_args_for_fermi.extend(['-qopenmp-simd', '-DSIMDE_ENABLE_OPENMP'])
+    elif new_gcc or clang or sysconfig.get_config_vars()['CC'] == 'clang':
+        extra_c_args_for_fermi.extend(['-fopenmp-simd', '-DSIMDE_ENABLE_OPENMP'])
```

Comment:
@mr-c I noticed that I wroted differently with your example:
https://github.com/biocore/scikit-bio/blob/2337b6a2bb79f0727b646f894bc5d92c5ca43c71/setup.py#L93

Should I not set `-fopenmp-simd -DSIMDE_ENABLE_OPENMP` when `CC` is `clang`?

Reply:
Last I knew, clang does not support ` -fopenmp-simd`

Reply:
@mr-c When I compiled the klib `ksw.c` file in `fermi-lite`, I didn't see any error from clang-12 in MacOSX:

```
$clang -v
Apple clang version 12.0.0 (clang-1200.0.32.27)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
$clang -c -g -Wall -O2 -Wno-unused-function -DNO_WARN_X86_INTRINSICS   -DUSE_SIMDE -DSIMDE_ENABLE_NATIVE_ALIASES -DSIMDE_ENABLE_OPENMP -fopenmp-simd   -I. ksw.c -o ksw.o
$file ksw.o
ksw.o: Mach-O 64-bit object x86_64
```
No complaints from clang-11 in conda in our linux server as well:
```
$clang -v
clang version 11.0.0 (https://github.com/conda-forge/clangdev-feedstock 234a25ce90b797cf04427a62cd668fc985808c23)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /mnt/lustre/users/tliu/miniconda3/bin
Found candidate GCC installation: /mnt/lustre/users/tliu/miniconda3/bin/../lib/gcc/x86_64-conda-linux-gnu/8.4.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-redhat-linux/4.8.2
Found candidate GCC installation: /usr/lib/gcc/x86_64-redhat-linux/4.8.5
Selected GCC installation: /mnt/lustre/users/tliu/miniconda3/bin/../lib/gcc/x86_64-conda-linux-gnu/8.4.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
$clang -c -g -Wall -O2 -Wno-unused-function -DNO_WARN_X86_INTRINSICS   -DUSE_SIMDE -DSIMDE_ENABLE_NATIVE_ALIASES -DSIMDE_ENABLE_OPENMP -fopenmp-simd   -I. ksw.c -o ksw.o
$file ksw.o
ksw.o: ELF 64-bit LSB relocatable, x86-64, version 1 (SYSV), not stripped
```